File: frontend/src/common/logger.ts
import {log} from "../config.ts";


export class Logger {
    static logError(error_text: string): void {
        log.error(`ERROR: ${this.getCurrentData()}: ${error_text}`)
    }

    static logResponse(response_text: string): void {
        log.info(`RES: ${this.getCurrentData()}: ${response_text}`)
    }

    static logRequest(request_text: string): void {
        log.info(`REQ: ${this.getCurrentData()}: ${request_text}`)
    }

    static getCurrentData(): string {
        return new Date().toISOString();
    }
}

File: backend/common/logger.py
import logging
from datetime import datetime

# logger for the application
logger = logging.getLogger('myapp')


class Logger:
    @staticmethod
    def log_response(res_str: str = ""):
        logger.info(f'INFO: {Logger.get_current_data_time()}: {res_str}')

    @staticmethod
    def log_error(error_str: str = ""):
        logger.error(f'ERROR: {Logger.get_current_data_time()}: {error_str}')

    @staticmethod
    def log_request(request_str: str = ""):
        logger.info(f'INFO: {Logger.get_current_data_time()}: {request_str}')

    @staticmethod
    def get_current_data_time() -> str:
        return datetime.now().strftime("%Y-%m-%d %H:%M:%S")
